### PR TITLE
RSDK-5432 Update replay movement sensor to handle new and old data formats

### PR DESCRIPTION
**Pull request by vijayvuyyuru**
Hey yall, as part of [DATA-1887](https://viam.atlassian.net/browse/DATA-1887), the format of collected movement sensor data changed in a way that would break the parsing of a replay IMU. This pr fixes that and remains backwards compatible by handling both types. After migrating to the new formats fully, the old code can be removed(maybe a new-eng ticket).

**Testing**

I tested this by configuring a replay IMU based on data captured in the new format and verifying no errors appeared in the logs and I was able to see matching IMU data in app.viam.com

[DATA-1887]: https://viam.atlassian.net/browse/DATA-1887?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by vijayvuyyuru** [comment]
@kkufieta This should make odometry work with old and new format data. Just wanted to bring to your attention.

File: components/movementsensor/replay/replay.go [hunk 1]
```diff
@@ -259,6 +269,14 @@ func (replay *replayMovementSensor) AngularVelocity(ctx context.Context, extra m
 		return spatialmath.AngularVelocity{}, err
 	}
 
+	if isNewAngularVelocityFormat(data) {
+		angularStruct := data.GetFields()["linear_velocity"].GetStructValue()
```
Comment (tessavitabile) [inline components/movementsensor/replay/replay.go:273]: "linear_velocity" should be "angular_velocity"

**Review by tessavitabile — APPROVED** [review]
LGTM with one requested change. Thank you for doing this! Also, would you be willing to file the ticket for removing support for the old format?

**Comment by vijayvuyyuru** [comment]
@tessavitabile https://viam.atlassian.net/browse/RSDK-5448 
Here is the ticket. I added the `new-eng` label to it and blocked it on the migration to the new data format. Let me know if you'd like me to change anything on it.

**Comment by tessavitabile** [comment]
Thank you for the fix, and for filing the cleanup ticket!

**Review by jeremyrhyde — COMMENTED** [review]
Changes look good, but what is the change to the collector for replay sensors? It seems like data will be returned as a single struct, accounting for all endpoints?

**Comment by vijayvuyyuru** [comment]
Chatted offline with @jeremyrhyde and answered his question. Functionally, nothing changes except being able to handle the new data format.

**Pull request merged**